### PR TITLE
tso, metrics: add the DC label for TSO metrics

### LINKED ISSUE #3238: Add some metrics to show the Local TSO information
### Describe your feature request related problem

Now the PD cluster supports to allocate the Local TSO, however, we don't have any metric to show the related information.

### Describe the feature you'd like

Add some metrics to show the Local TSO information so that we can monitor the Local TSO things from the Grafana.

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Close #3238.

### What is changed and how it works?

* Add the DC label for TSO metrics to better distinguish between TSO metrics of different DCs.
* Fix the bug causes PD role info is not accurate.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

**Comment by rleungx** [comment]
Do we need to modify the Grafana template?

**Comment by JmPotato** [comment]
> Do we need to modify the Grafana template?

Yes, I will update it later.

**Comment by JmPotato** [comment]
图片
图片

Now the PD role info will be more accurate.

**Comment by JmPotato** [comment]
@rleungx @nolouch PTAL

**Review by nolouch — COMMENTED** [review]
rest LGTM

**Comment by JmPotato** [comment]
> rest LGTM

Does it seem that the review content is missed?

File: metrics/grafana/pd.json [hunk 1]
```diff
@@ -134,8 +134,9 @@
       "tableColumn": "",
       "targets": [
         {
-          "expr": "delta(pd_tso_events{tidb_cluster=\"$tidb_cluster\", type=\"save\",instance=\"$instance\"}[1m]) > bool 0",
+          "expr": "pd_cluster_tso{tidb_cluster=\"$tidb_cluster\", instance=\"$instance\", dc=\"global\"} > bool 0",
```
Comment (JmPotato) [inline metrics/grafana/pd.json:137]: It seems that this PromoQL is still not accurate here.

File: metrics/grafana/pd.json [hunk 2]
```diff
@@ -134,8 +134,9 @@
       "tableColumn": "",
       "targets": [
         {
-          "expr": "delta(pd_tso_events{tidb_cluster=\"$tidb_cluster\", type=\"save\",instance=\"$instance\"}[1m]) > bool 0",
+          "expr": "pd_tso_role{tidb_cluster=\"$tidb_cluster\", instance=\"$instance\", dc=\"global\"}",
```
Comment (nolouch) [inline metrics/grafana/pd.json:137]: This should be incompatible with metrics without global txn enabled, will it cause data to be unavailable under normal tso?

**Review by nolouch — COMMENTED** [review]
lgtm

File: metrics/grafana/pd.json (hunk 2 above)
Comment (JmPotato) [reply metrics/grafana/pd.json:137]: It's compatible without Global TSO enabled.

Comment (JmPotato) [reply metrics/grafana/pd.json:137]: Normal TSO and Global TSO are the same things now. Global TSO will behave differently according to the cluster state.

Comment (nolouch) [reply metrics/grafana/pd.json:137]: you mean If not enabled Global TSO, dc also set to `global`?

Comment (JmPotato) [reply metrics/grafana/pd.json:137]: Yes.

**Review by nolouch — APPROVED** [review]
lgtm

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**